File: demos/jans-chip/ios/Jans-Chip/Models/Networking/AttestationOptionResponse.swift
//
//  AttestationOptionResponse.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 11.07.2024.
//

import Foundation

public struct AttestationOptionResponse: Codable, ErrorHandler {
    var errorMessage: String?
    var isSuccess: Bool = true
    
    private enum CodingKeys: String, CodingKey {
        case attestation
        case authenticatorSelection
        case challenge
        case pubKeyCredParams
        case rp
        case user
    }
    
    var attestation: String?
    var authenticatorSelection: AuthenticatorSelection?
    var challenge: String?
    var pubKeyCredParams: [PubKeyCredParam]?
    var rp: RP?
    var user: User?
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/fido/attestation/option/AttestationOptionResponse.kt
package io.jans.chip.model.fido.attestation.option

import androidx.room.Ignore

class AttestationOptionResponse (
     var attestation: String?,
     var authenticatorSelection: AuthenticatorSelection?,
     var challenge: String?,
     var pubKeyCredParams: List<PubKeyCredParam>?,
     var rp: RP?,
     var user: User?,
) {
     @Ignore
     var isSuccessful: Boolean? = true

     @Ignore
     var errorMessage: String? = null
}